File: daily/2025/1295_Find_Numbers_with_Even_Number_of_Digits.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    int findNumbers(vector<int>& nums) {
        int ans = 0;
        for (int num : nums){
            int count = 0;
            while(num){
                count++;
                num /= 10;
            }
            if (count % 2 == 0) ans++;
        }
        return ans;
    }
};

int main(){
    vector<int> nums = {12,345,2,6,7896};
    // vector<int> nums = {555,901,482,1771};
    Solution S;
    cout << S.findNumbers(nums) << endl;
    return 0;
}

File: daily_python/1295_Find_Numbers_with_Even_Number_of_Digits.py
import os
import sys
from typing import List

class Solution:
    def findNumbers(self, nums: List[int]) -> int:
        ans = 0
        for num in nums:
            count = 0
            while num > 0:
                count += 1
                num //= 10
            if count % 2 == 0:
                ans += 1
        return ans
    
def readDataSet(filename):
    dataset = []
    with open(filename, 'r') as file:
        content = file.read().strip()
        blocks = content.split('\n\n')
        for block in blocks:
            lines = block.split('\n')
            nums = list(map(int, lines[0].split('=')[1].strip()[1:-2].split(',')))
            dataset.append(nums)
    return dataset

if __name__ == '__main__':
    if len(sys.argv) == 1:
        filename = os.path.splitext(__file__)[0] + '.txt'
    else:
        filename = sys.argv[1]
    dataset = readDataSet(filename)
    results = []
    solution = Solution()
    for nums in dataset:
        results.append(solution.findNumbers(nums))
    for index, result in enumerate(results):
        print(f'Example {index + 1} : {result}')